src/main/java/ay2122s1_cs2103t_w16_2/btbb/model/ingredient/UniqueIngredientList.java

```diff
@@ -71,6 +72,30 @@ public void setIngredients(List<Ingredient> ingredients) {
         internalList.setAll(ingredients);
     }
 
+    /**
+     * Replaces the similar ingredient that is in the address book with a new ingredient whose quantity is reduced
+     * by the quantity times the multiplier in {@code target} if it exists.
+     *
+     * @param target The target ingredient.
+     * @param multiplier The multiplier.
+     */
+    public void minusIngredientQuantity(Ingredient target, GenericQuantity multiplier) {
+        requireAllNonNull(target, multiplier);
+
+        Ingredient currentIngredient = internalList.stream()
+                .filter(target::isSameIngredient).findFirst().orElse(null);
+
+        if (currentIngredient != null) {
+            Ingredient ingredientWithNewQuantity = new Ingredient(
+                    currentIngredient.getName(),
+                    currentIngredient.getQuantity().minusQuantityBy(target.getQuantity(), multiplier),
+                    currentIngredient.getUnit());
+
+            int index = internalList.indexOf(currentIngredient);
+            internalList.set(index, ingredientWithNewQuantity);
+        }
```

Comment:
Maybe can extract out the part where you set the existing ingredient with a new ingredient? Since it would be reused later for edit ingredient
```suggestion
    public void setIngredient(Ingredient target, Ingredient editedIngredient) throw NotFoundException {
        requireAllNonNull(target, editedIngredient);
        int index = internalList.indexOf(target);
        if (index == -1) {
              throw new NotFoundException(Ingredient.class.getName());
        }
        internalList.set(index, editedIngredient);
    }

    public void minusIngredientQuantity(Ingredient target, GenericQuantity multiplier) {
        requireAllNonNull(target, multiplier);

        Ingredient currentIngredient = internalList.stream()
                .filter(target::isSameIngredient).findFirst().orElse(null);

        if (currentIngredient != null) {
            Ingredient ingredientWithNewQuantity = new Ingredient(
                    currentIngredient.getName(),
                    currentIngredient.getQuantity().minusQuantityBy(target.getQuantity(), multiplier),
                    currentIngredient.getUnit());

            setIngredient(target, ingredientWithNewQuantity);
        }
```

Reply:
I think there is no need to extract. The `setIngredient` throws a `NotFoundException` which i have to catch in `minusIngredientQuantity` because the goal is to do nothing if the ingredient is not found and only minus for ingredients that are in the list. So there will be empty catch blocks which is not good coding practice.